### ISSUE TITLE
cannot capture arrow keys

### ISSUE THREAD

**Issue by hozen** [pyserial/pyserial#207]
I ran a Console().getkey() in a while loop, and while pressing arrow keys, for example UP arrow, I can only get 27 and 91 (in ASCII code)  but lack of 65. here 27, 91, 65 means "^[[A", that is corresponding to arrow UP key.

And only after pressing "Enter" key, the 65 appears.

**Comment by zsquareplusc** [comment]
Check that you have the latest version with this fix: cab3dabc81ecbf010d81ff38433db99c1f6d667c

There was a problem in a recent version where `select` was used to implement the `cancel` function. The combination of `codec` and `select` caused blocking reads while there was still data.

Please also note that special keys are not treated specially and so behave differently under POSIX and Windows. I have a patched version here, that also returns escape sequences on Windows (at least for some common keys): [miniterm_mpy.py](https://github.com/zsquareplusc/mpy-repl-tool/blob/master/there/miniterm_mpy.py)

**Comment by hozen** [comment]
@zsquareplusc , yes, i have the latest version code, and I also noticed that issue prior to the fix made. But the issue I reported in this post is still existed. I have read your improvement for special keys, but it only implemented on NT os, would you mind add it to POSIX either? Thanks in advance.

**Comment by zsquareplusc** [comment]
the translation of keys is only required on NT as under POSIX you usually already have a capable terminal that sends these escape sequences.
